src/Data/Text/Markdown.hs

```diff
@@ -0,0 +1,268 @@
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.Text.Markdown (
+  -- ** Markdown document
+  Document (..),
+  Paragraph (..),
+  Node (..),
+  TxtAttr (..),
+  fromTextM,
+
+  -- ** Token stream
+  StreamNode' (..),
+  StreamNode,
+  ToStream (..),
+  toText,
+
+  -- ** Utilities
+  findCode,
+  chunksOf,
+) where
+
+import Commonmark qualified as Mark
+import Commonmark.Extensions qualified as Mark (rawAttributeSpec)
+import Control.Applicative ((<|>))
+import Control.Arrow (left)
+import Control.Monad (void)
+import Data.Functor.Identity (Identity (..))
+import Data.List qualified as List
+import Data.List.Split (chop)
+import Data.Maybe (catMaybes)
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Text (Text)
+import Data.Text qualified as T
+import Data.Tuple.Extra (both, first)
+import Data.Vector (toList)
+import Data.Yaml
+import Swarm.Language.Module (moduleAST)
+import Swarm.Language.Parse (readTerm)
+import Swarm.Language.Pipeline (ProcessedTerm (..), processParsedTerm)
+import Swarm.Language.Pretty (prettyText, prettyTypeErrText)
+import Swarm.Language.Syntax (Syntax)
```

Comment:
Depending on Swarm in Data module is a hack. 🤷  Suggestions welcome for better placement or architecture.

Reply:
Yeah, our use case is specifically inspecting swarm code in descriptions, so generalising does not make that much sense.